hikari/files.py

```diff
@@ -473,6 +473,38 @@ async def read(
         async with self.stream(executor=executor) as reader:
             return await reader.read()
 
+    async def save(
+        self,
+        *,
+        path: typing.Optional[typing.Union[pathlib.Path, str]] = None,
+        executor: typing.Optional[concurrent.futures.Executor] = None,
+    ) -> None:
+        """Save this resource to disk.
+
+        This writes the resource file in chunks, and so does not load
+        the entire resource into memory.
+
+        Parameters
+        ----------
+        path : typing.Optional[typing.Union[pathlib.Path, str]]
+            The path to save this resource to. If this is a string, the
+            path will be relative to the current working directory. If
+            `builtins.None`, the resource will be saved in the
+            current working directory with the same name as it has on
+            Discord.
```

Comment:
```suggestion
            current working directory with the stored filename.
```

Reply:
I think the original wording is better here. The meaning of "stored filename" is not particularly clear.

Reply:
The original wording is not really correct, as (for example) `URL` doesn't come from Discord. "stored filename" was refering to the `filename` attribute. Instead could just say that:

```suggestion
            current working directory under name in `filename`.
```

Reply:
That's fair.